src/Contracts/Services/ShoppingCart/Query.cs

```diff
@@ -1,18 +1,43 @@
 ﻿using Contracts.Abstractions.Messages;
+using Contracts.Abstractions.Paging;
 
 namespace Contracts.Services.ShoppingCart;
 
 public static class Query
 {
-    public record GetShoppingCart(Guid CartId) : Message, IQuery;
+    public record GetShoppingCartDetails(Guid CartId) : IQuery
+    {
+        public static implicit operator GetShoppingCartDetails(Protobuf.GetShoppingCartDetailsRequest request)
+            => new(new Guid(request.CartId));
```

Comment:
Here is possible use this strategy.

```suggestion
            => new(new(request.CartId));
```